server/lib/notifications.js

```diff
@@ -215,9 +215,87 @@ const notifyUpdateSubscribers = async activity => {
   const emailOpts = { from: activity.data.fromEmail };
   const update = await models.Update.findByPk(activity.data.update.id);
   const allUsers = await update.getUsersToNotify();
-  return notifySubscribers(allUsers, activity, emailOpts);
+  const modifiedActivity = replaceVideosByImagePreviews(activity);
+  return notifySubscribers(allUsers, modifiedActivity, emailOpts);
 };
 
+function replaceVideosByImagePreviews(activity) {
+  const regexImg = new RegExp(`<iframe([\\w\\W]+?)[\\/]?>`, 'g');
+  const regexUrl = new RegExp(
+    `(https):\\/\\/([\\w_-]+(?:(?:\\.[\\w_-]+)+))([\\w.,@?^=%&:/~+#-]*[\\w@?^=%&/~+#-])?`,
+    'ig',
+  );
+  activity.data.update.html = activity.data.update.html
+    .replace(regexImg, match => {
+      let imageReplacement;
+      if (match.includes('youtube')) {
+        imageReplacement = match
+          .replace(new RegExp('<iframe', 'g'), '<img')
+          .replace(new RegExp('allowfullscreen', 'g'), '')
+          .replace(new RegExp('frameborder="0"', 'g'), '');
+
+        const { videoService, videoId } = getEmbedDetails(imageReplacement);
+        const previewImageURL = constructPreviewImageURL(videoService, videoId);
+        return imageReplacement.replace(regexUrl, previewImageURL);
+      }
+    })
+    .replace(new RegExp('<figure data-trix-content-type="([\\w\\W]+?)[\\/]?">', 'g'), '')
+    .replace(new RegExp('</iframe><figcaption></figcaption></figure>', 'g'), '')
+    .replace(new RegExp('width="100%" height="394"', 'g'), '');
```

Comment:
It's quite complex to undestand this code without proper context. It also looks like `.replace(new RegExp('<figure data-trix-content-type="([\\w\\W]+?)[\\/]?">', 'g'), '')` can have side effects that are not related to iframes.

I haven't tested the following code (other than the regex with https://regex101.com/r/OiYxtK/1), but I think using a slightly more advanced regex for the iframe can really simplify things here:

```es6
const iframePreviewRegex = /<iframe\s[^>]*src="([^"]+)"[^>]*><\/iframe>/ig;
activity.data.update.html = activity.data.update.html.replaceAll(iframePreviewRegex, (_, href) => {
  const { service, id } = parseServiceLink(href);
  const imgSrc = constructPreviewImageURL(service, id);
  if (imgSrc) {
    return `<img src="${imgSrc}" alt="${service} content">`
  } else {
    return ''
  }
})
```

Reply:
@Betree : Thanks much. I've refactored this based on your suggestion. Seems to work perfectly. Now as for the `.replace(new RegExp('<figure data-trix-content-type="([\\w\\W]+?)[\\/]?">', 'g'), '')`, yes you are right; this is an oversight on my part and it could have side effects. 🙄 I removed this line as it's really not needed to remove these `<figure` tags anyways. 🐨